src/js/components/DataTable/DataTable.js

```diff
@@ -173,6 +176,40 @@ const DataTable = ({
     }
   }, [footerRef, headerRef, placeholder]);
 
+  useEffect(() => {
+    const getPinnedOffset = () => {
+      const pinnedProperties = columns
+        .map(pinnedColumn => pinnedColumn.pin && pinnedColumn.property)
+        .filter(n => n);
+
+      const pinnedColumnsProperties = {};
+      const firstRowCells = headerRef?.current?.firstChild?.cells || [];
+
+      if (firstRowCells.length !== 0) {
+        pinnedProperties.forEach((property, index) => {
+          const tableIndex = columns.findIndex(
+            column => column.property === property,
+          );
+
+          if (firstRowCells[tableIndex]) {
+            pinnedColumnsProperties[property] = {
+              width: firstRowCells[tableIndex].getBoundingClientRect()?.width,
+              left:
+                tableIndex === 0
+                  ? 0
+                  : pinnedColumnsProperties[pinnedProperties[index - 1]].left +
+                    pinnedColumnsProperties[pinnedProperties[index - 1]].width,
+            };
+          }
+        });
+      }
+      return pinnedColumnsProperties;
+    };
+
+    setPinnedOffset(getPinnedOffset());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can you add a comment about why we need to ignore exhaustive-deps warning? If dependency array is not desired, can we leave the `[]` out entirely?
```
useEffect(() => {...});
```

Reply:
We can try a workaround on that, but just removing the dependency array is leading to exhaustive-deps warnings. Probably because of line 209: `setPinnedOffset(getPinnedOffset());`

![image](https://user-images.githubusercontent.com/47114840/115761118-5e64cf00-a378-11eb-8b0f-42af835cff1b.png)


We can also set `columns` as a dependency. Might be clever considering we want this to update if columns are updated as well. What do you think?

Reply:
Following up from offline discussion. Recommending we try `useMemo` for `pinnedOffset` instead to potentially overcome the 'maximum depths' issue.